SampleGallery/Samples/SDK 16299/LinearGradients/LinearGradients.xaml.cs

```diff
@@ -0,0 +1,263 @@
+        }
+
+        // That switches the sets of colors (DeepPink/Honeydew and LightSkyBlue/Teal)
+        private void SwitchColorSetOnPressed(CompositionColorGradientStop stop1, CompositionColorGradientStop stop2)
+        {
+            // If stop 1 is currently one of the warmer colors that are set together, it will call animations to switch both stops to the cooler color sets when the mouse clicks on the visuals
+            if (stop1.Color == _deepPink || stop1.Color == _honeydew)
+            {
+                stop1.StartAnimation("Color", _changeStopToLightSkyBlue);
+                stop2.StartAnimation("Color", _changeStopToTeal);
+            }
+            // If stop 1 is currently one of the cooler colors that are set together, it will call animations to switch both stops to the warmer color sets when the mouse clicks on the visuals
+            else if (stop1.Color == _lightSkyBlue || stop1.Color == _teal)
+            {
+                stop1.StartAnimation("Color", _changeStopToDeepPink);
+                stop2.StartAnimation("Color", _changeStopToHoneydew);
+            }
+        }
+
+        // Animates scale of visual that method is called on
+        private void AnimateScale(SpriteVisual target)
+        {
+            target.StartAnimation("Scale", _scaleAnim);
+        }
+
+        // Switch the current colors of the two color stops in the linear gradient brush. Called when first XAML UI Element is entered by the mouse pointer
+        private void ChangeStopColorsOnEntered(CompositionColorGradientStop stop1, CompositionColorGradientStop stop2)
+        {
+            if (stop1.Color == _deepPink)
+            {
+                stop1.StartAnimation("Color", _changeStopToHoneydew);
+                stop2.StartAnimation("Color", _changeStopToDeepPink);
+            } else if (stop1.Color == _teal)
+            {
+                stop1.StartAnimation("Color", _changeStopToLightSkyBlue);
+                stop2.StartAnimation("Color", _changeStopToTeal);
+            }
+        }
+
+        // Switch the current colors of the two color stops in the linear gradient brush. Called when last XAML UI Element is entered by the mouse pointer
+        private void ReverseStopColorsOnEntered(CompositionColorGradientStop stop1, CompositionColorGradientStop stop2)
+        {
+            if (stop1.Color == _honeydew)
+            {            
+                stop1.StartAnimation("Color", _changeStopToDeepPink);
+                stop2.StartAnimation("Color", _changeStopToHoneydew);
+            } else if (stop1.Color == _lightSkyBlue)
+            {
+                stop1.StartAnimation("Color", _changeStopToTeal);
+                stop2.StartAnimation("Color", _changeStopToLightSkyBlue);
+            }
+        }
+
+        // Animate the offset of the visuals to move back and forth the length of the given visual that this method is called on
+        private void AnimateOffset(SpriteVisual target)
```

Comment:
Looks like AnimateScale and AnimateOffset are always used together. Why not have a single method so that callers can make a single call instead of 2?

Reply:
i wanted my code to be incredibly readable so i separated these to help with that. i can put them together but i thought this was a good decision to make my code easier for those not familiar with composition - if they wanted to only understand the offset animation they could follow that thread without it being confusing and adding in the scale animation.